extension/parquet/parquet_extension.cpp

```diff
@@ -1390,6 +1396,58 @@ unique_ptr<TableRef> ParquetScanReplacement(ClientContext &context, ReplacementS
 	return std::move(table_function);
 }
 
+static BoundStatement WriteGeoParquetPlan(Binder &binder, CopyStatement &stmt) {
+
+	// First off, check if we even have the spatial extension installed
+	if (!GeoParquetFileMetadata::IsSpatialExtensionInstalled(binder.context)) {
+		// Just fall back to regular parquet if we don't have the spatial extension
+		stmt.info->format = "parquet";
+		return binder.Bind(*stmt.info->select_statement);
+	}
+
+	auto stmt_copy = stmt.Copy();
+	auto &copy = stmt_copy->Cast<CopyStatement>();
+	auto &copied_info = *copy.info;
+
+	auto dummy_binder = Binder::CreateBinder(binder.context, &binder);
+	auto bound_original = dummy_binder->Bind(*stmt.info->select_statement);
+
+	// Create new SelectNode with the original SelectNode as a subquery in the FROM clause
+	// We need to use a subquery here because the original SelectNode might be a star expression, so we can't just
+	// replace the expressions in the original SelectNode. We need to use positional references instead that get
+	// resolved during the second binding.
+
+	auto select_stmt = make_uniq<SelectStatement>();
+	select_stmt->node = std::move(copied_info.select_statement);
+	auto subquery_ref = make_uniq<SubqueryRef>(std::move(select_stmt));
+
+	copied_info.select_statement = make_uniq_base<QueryNode, SelectNode>();
+	auto &select_node = copied_info.select_statement->Cast<SelectNode>();
+	select_node.from_table = std::move(subquery_ref);
+
+	auto &catalog = Catalog::GetSystemCatalog(binder.context);
+	auto geometry_type = LogicalType(LogicalTypeId::BLOB);
+	geometry_type.SetAlias("GEOMETRY");
+
+	for (idx_t col_idx = 0; col_idx < bound_original.types.size(); col_idx++) {
```

Comment:
Can we perhaps extend `copy_supports_type_t` to insert the casts instead of doing the plan replacement?